src/psyclone/tests/psyir/transformations/loop_fusion_test.py

```diff
@@ -146,6 +146,36 @@ def fuse_loops(fortran_code, fortran_reader, fortran_writer):
     return fortran_writer(psyir), psyir
 
 
+# ----------------------------------------------------------------------------
+def test_fuse_dependency_tools_called(fortran_reader):
+    '''Make sure that DepencendyTools.can_loops_be_fused is indeed called
+    from the loop fuse transformation with the right parameters.
+    '''
+
+    # First test: read/read of scalar variable
+    code = '''subroutine sub()
+              integer :: jj, n
+              real, dimension(10) :: s, t
+              real                :: a
+              do jj=1, n
+                 s(jj) = t(jj) + a
+              enddo
+              do jj=1, n
+                 t(jj) = t(jj) - a
+              enddo
+              end subroutine sub'''
+    psyir = fortran_reader.psyir_from_source(code)
+    # Get the PSyIR of the original loops
+    loop1 = psyir.children[0].children[0]
+    loop2 = psyir.children[0].children[1]
+
+    fuse = LoopFuseTrans()
```

Comment:
You could monkeypatch `DependencyTools.can_loops_be_fused` for this test instead?